lib/common/options.c

```diff
@@ -1041,6 +1059,11 @@ pcmk__format_option_metadata(const char *name, const char *desc_short,
         const char *opt_desc_short = option_list[lpc].description_short;
         const char *opt_desc_long = option_list[lpc].description_long;
 
+        if ((filter != pcmk__opt_context_none)
+            && (filter != option_list[lpc].context)) {
```

Comment:
We can switch this to `pcmk_is_set(option_list[lpc].context, filter)` in the future if any options have multiple context flags set. For now, an option holds a single enum value rather than `uint32_t` flag set., so we use this notation for clarity.